gocontext.go

```diff
@@ -7,23 +7,17 @@ type contextKey struct{}
 var activeSpanKey = contextKey{}
 
 // ContextWithSpan returns a new `context.Context` that holds a reference to
-// the given `Span`.
+// the given `Span`'s `SpanContext`.
 func ContextWithSpan(ctx context.Context, span Span) context.Context {
```

Comment:
Since the proposed semantics allow for `Span.SpanContext()` to be called after `Span.Finish()`, I think it's fine to store the `Span` in the `context.Context